nixos/modules/tasks/filesystems.nix

```diff
@@ -155,6 +174,17 @@ in
       description = "Names of supported filesystem types.";
     };
 
+    boot.specialFileSystems = mkOption {
+      type = types.listOf (types.submodule coreFileSystemOpts);
+      apply = makeSpecialMounts;
```

Comment:
Don't use `apply`, it's ugly. If you really must generate a shell script (rather than say a file in /etc/fstab format), it's better to have a value like `sytem.build.specialFilesystemMountScript` derived from `boot.specialFileSystems`.